[Docs] [Console] Fixed/updated documentation links in Dev Tools console

curq opened the pull request:
### Description
1. After auditing JSON files in https://github.com/opensearch-project/OpenSearch-Dashboards/tree/main/src/plugins/console/server/lib/spec_definitions/json/generated I updated every old link to the relevant link to documentation. They were outdated, so every link was updated. Now in dev console they open correct links instead of outdated ones that redirect to the main page.
2. Every api endpoint that wasn't documented in docs was listed in the new issue as pre request https://github.com/opensearch-project/documentation-website/issues/3614 Also, links were updated to `https://opensearch.org/docs/`
3. The https://github.com/opensearch-project/OpenSearch-Dashboards/tree/main/src/plugins/console/server/lib/spec_definitions/js/ingest.ts was also audited and links were updated.
4. Many processors were not documented, so their links were changed to the `https://opensearch.org/docs/latest/data-prepper/pipelines/configuration/processors/processors/` and reported missed documentation in https://github.com/opensearch-project/documentation-website/issues/3614
5. As per exit criteria no strings matching `/guide/en/elasticsearch` were left in the plugin directory.

Important notes:
- info.json and ping.json had link that pointed to `https://opensearch.org/docs/latest/opensearch/index/`, which I assumed was base url so they were changed to  `https://opensearch.org/docs/`
- in get_script_languages.json and get_script_context.json there was no documentation attribute so the Open Documentation button in Dev Tools console was not working, so I added it with appropriate link and now it opens correct documentation page.
- cluster.delete_component_template.json, cluster.exists_component_template.json, cluster.get_component_template.json were not documented but instead of main docs page I linked them to `https://opensearch.org/docs/latest/im-plugin/index-templates/`, because PUT method of the same pattern was documented there so I assumed that after docs will be updated this will be the correct link.
- same situation with reindex_rethrottle.json, delete_by_query_rethrottle.json and update_by_query_rethrottle.json. I linked them to the same links as reindex.json, delete_by_query.json and update_by_query.json. It is the same pattern as with the outdated links, because I presume that rethrottle variation of pattern should be described on the same page. At least I found similar pattern in elasticsearch documentation of the same features. (e.g. https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-reindex.html)

### Issues Resolved
 Fixes #3714
### Check List
- [ ] All tests pass
  - [X] `yarn test:jest`
  - [X] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [X] Update [CHANGELOG.md](./../CHANGELOG.md)
- [X] Commits are signed per the DCO using --signoff

AMoo-Miki commented:
Thanks Sirazh for this.

joshuarrrr commented:
@curq This is awesome - I'm excited to get it in the next release! I'm reviewing today and should have some feedback shortly.

joshuarrrr reviewed (commented):
@curq Thanks, this is excellent - great job finding the most relevant doc available. I was able to find just a couple more, and in some cases, even where we don't have docs available, I think we can at least get the user closer, so I made some suggestions along those lines.

joshuarrrr commented:
Failing cypress test workflow can be ignored

curq commented on src/plugins/console/server/lib/spec_definitions/json/generated/indices.exists_template.json:
```diff
@@ -10,6 +10,6 @@
       "_template/{name}"
     ],
-    "documentation": "https://opensearch.org/docs/latest/opensearch/index-templates/"
+    "documentation": "https://opensearch.org/docs/latest/im-plugin/index-templates/"
```
At first I also wanted to use the suggested link, but then I noticed that pattern in Index exists `HEAD /<index-name>` is not similar to the _template/{name}, while the index-templates page have example with  basically identical pattern. The following screenshot is from the index-templates page:
<img width="370" alt="image" src="https://user-images.githubusercontent.com/67782303/228705228-cb801da7-d204-4670-82f8-1cc0b9d79571.png">

joshuarrrr replied:
yep, that sounds good

curq commented on src/plugins/console/server/lib/spec_definitions/json/generated/snapshot.clone.json:
```diff
@@ -9,6 +9,6 @@
       "_snapshot/{repository}/{snapshot}/_clone/{target_snapshot}"
     ],
-    "documentation": "https://opensearch.org/docs/latest/opensearch/rest-api/"
+    "documentation": "https://opensearch.org/docs/"
```
I agree that it is the closest docs page we currently have, but I hesitated to include this link because there was no `_clone` pattern mentioned there. If that is fine I will do as suggested.

joshuarrrr replied:
Yeah - we'll need to do a separate audit to bring better alignment between the methods supported in the console and what's actually documented on the OpenSearch side.

curq commented on src/plugins/console/server/lib/spec_definitions/json/generated/snapshot.cleanup_repository.json:
```diff
@@ -10,6 +10,6 @@
       "_snapshot/{repository}/_cleanup"
     ],
-    "documentation": "https://opensearch.org/docs/latest/opensearch/snapshot-restore/"
+    "documentation": "https://opensearch.org/docs/"
```
I checked the previous link, it is broken and does not point to existing page so it just redirects to the main page.  I will update it to the fixed https://opensearch.org/docs/latest/tuning-your-cluster/availability-and-recovery/snapshots/snapshot-restore/ link.

curq commented on src/plugins/console/server/lib/spec_definitions/json/generated/indices.get_mapping.json:
```diff
@@ -20,6 +20,6 @@
       "{indices}/_mapping"
     ],
-    "documentation": "https://opensearch.org/docs/latest/guide/en/elasticsearch/reference/master/indices-get-mapping.html"
+    "documentation": "https://opensearch.org/docs/latest/field-types/mappings/"
```
I think it is better resource because get mapping is directly documented on this page. Here is the screenshot 
<img width="745" alt="image" src="https://user-images.githubusercontent.com/67782303/228707754-b7dca1ca-7a53-4df9-a381-97865bf52523.png">

curq replied:
I will update it to the link with anchor so that it will automatically scroll to the required part of the page.
https://opensearch.org/docs/latest/field-types/mappings/#get-a-mapping

curq commented on src/plugins/console/server/lib/spec_definitions/json/generated/cluster.delete_voting_config_exclusions.json:
```diff
@@ -9,6 +9,6 @@
       "_cluster/voting_config_exclusions"
     ],
-    "documentation": "https://opensearch.org/docs/latest/opensearch/rest-api/"
+    "documentation": "https://opensearch.org/docs/"
```
Got it, I will change them.

curq commented:
Thank you for your review @joshuarrrr 
I answered to some of your comments, can you please look through them? Then I will update the links according to the proposed changes.

Also I have a question, if I commit and push the new changes to my remote fork branch will they appear in this pull request or is there another workflow?

seanneumann commented:
This is amazing! Thanks @curq!!

curq commented on src/plugins/console/server/lib/spec_definitions/json/generated/info.json:
```diff
@@ -6,6 +6,6 @@
       ""
     ],
-    "documentation": "https://opensearch.org/docs/latest/opensearch/index/"
+    "documentation": "https://opensearch.org/docs/"
```
It believe it was previously pointing at the main docs page, which I think is reasonable if we consider pattern ""
<img width="647" alt="image" src="https://user-images.githubusercontent.com/67782303/228841456-ddc8dace-f7e9-4263-b497-b28828d834fc.png">
So I believe updating this link to https://opensearch.org/docs/ will be more appropriate? In this case when no command is written (pattern "") it will point to the index page of opensearch docs.

curq commented:
I added some more changes after reviewing the feedback here is short summary of changes:
- Updated `cluster.*` missing links to https://opensearch.org/docs/latest/api-reference/cluster-api/index/
- Updated `indices.*` missing links to https://opensearch.org/docs/latest/api-reference/cluster-api/index/ I also updated other similar `indices.*` files that were not mentioned in feedback as I noticed them after rechecking the json files.
- Updated links with anchor tags to make user jump to the relevant section of the page immediately.
- Updated links in .json files that were suggested by feedback like cat.master.json or ingest.processor_grok.json
- Fixed CHANGELOG.md formatting by adding the space as was suggested.

In the suggestions above I commented some of my reasoning behind the changes.

zhongnansu commented on CHANGELOG.md:
```diff
@@ -142,6 +142,7 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 - [Doc] Add docker dev set up instruction ([#3444](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3444))
 - [Doc] UI actions explorer ([#3614](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3614))
 - [Doc] Update SECURITY.md with instructions for nested dependencies and backporting ([#3497](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3497))
+- [Doc] [Console] Fix/update documentation links in Dev Tools console ([#3714](https://github.com/opensearch-project/OpenSearch-Dashboards/issues/3714))
```
```suggestion
- [Doc] [Console] Fix/update documentation links in Dev Tools console ([#3724](https://github.com/opensearch-project/OpenSearch-Dashboards/issues/3724))
```

curq replied:
Done!

zhongnansu reviewed (commented):
Thanks for the change!
Minor typo on change log, plz fix it and we are good to go

curq commented:
@joshuarrrr @zhongnansu Fixed typo in changelog, ready for approval

zhongnansu reviewed (approved):
ShIp it!

joshuarrrr commented:
All tests pass, I'll resolve the CHANGELOG conflict.

joshuarrrr commented:
backporting wasn't completed in time for 1.3.10 - relabeling for 1.3.11.
